src/components/Modals/Receive/ReceiveInfo.tsx

```diff
@@ -65,20 +66,26 @@ export const ReceiveInfo = ({ asset, accountId }: ReceivePropsType) => {
   const bip44Params = accountMetadata?.bip44Params
   useEffect(() => {
     ;(async () => {
-      if (!(wallet && chainAdapter)) return
-      if (!bip44Params) return
-      // if (chainAdapter.isAccountTypeRequired() && !accountType) return
-      const { chainNamespace } = fromChainId(asset.chainId)
-      if (CHAIN_NAMESPACE.Utxo === chainNamespace && !accountType) return
-      const { accountNumber } = bip44Params
-      const selectedAccountAddress = await chainAdapter.getAddress({
+      if (!accountMetadata) return
+      if (!wallet) return
+      const selectedAccountAddress = await getReceiveAddress({
+        asset,
         wallet,
-        accountType,
-        accountNumber,
+        deviceId: await wallet.getDeviceID(),
+        accountMetadata,
```

Comment:
not introduced in this pr, but do we want loading state for this?

Reply:
We have a skeleton on l.163 which acts as a loading state on `Boolean(receiveAddress)` so I believe we don't need an additional field! https://github.com/shapeshift/web/blob/9f7a61918ad443553504edd2f7e6a374f316347b/src/components/Modals/Receive/ReceiveInfo.tsx#L163